### ISSUE TITLE
position: sharp.strategy.entropy not working, no errors, just get center position in output

### ISSUE THREAD

**Issue by davidrinnan** [lovell/sharp#3664]
### Is this a possible bug in a feature of sharp, unrelated to installation?

- [x] Running `npm install sharp` completes without error.
- [x] Running `node -e "require('sharp')"` completes without error.

If you cannot confirm both of these, please open an [installation issue](https://github.com/lovell/sharp/issues/new?labels=installation&template=installation.md) instead.

### Are you using the latest version of sharp?

- [x] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

If you cannot confirm this, please upgrade to the latest version and try again before opening an issue.

If you are using another package which depends on a version of `sharp` that is not the latest, please open an issue against that package instead.

### What are the steps to reproduce?
Lambda node 18 x86 built on x86 macos with the following: 
npm install
rm -rf node_modules/sharp
SHARP_IGNORE_GLOBAL_LIBVIPS=1 npm install --arch=x64 --platform=linux --libc=glibc sharp

    sharp(data.Body)
        .resize({
            width: 200,
            height: 200,
            fit: sharp.fit.cover,
            position: sharp.strategy.entropy
    }).toBuffer(function(err, resizedImage, info) {..................

### What is the expected behaviour?
We are doing a number of different conversions with sharp. All image operations completes 100% successfully BUT the position: sharp.strategy.entropy does not work. We have tried exchanging with the constant 16 but no success. We get no errors. We simply get a cropped image with position center. 

We expect entropy based cropping.

**Comment by lovell** [comment]
There are unit tests for this feature that are passing so it may be specific images that are failing. Please can you provide a sample image that exhibits this problem.

https://github.com/lovell/sharp/blob/5255964c79b6145650b48c5ee6e96b3e1094b9be/test/unit/resize-cover.js#L299

Do you see the `cropOffsetLeft` and `cropOffsetTop` properties of `info` in the callback?

**Comment by davidrinnan** [comment]
This image is one of the ones Ive tested on. I have tried the different strategies with the same result. always a top - left crop. I will go ahead and log the info properties mentioned above. 

![IMG_4098](https://github.com/lovell/sharp/assets/95898/c446258b-cc15-439d-a982-db33fb08086d)

**Comment by davidrinnan** [comment]
format: 'jpeg',
  width: 1000,
  height: 1000,
  channels: 3,
  premultiplied: false,
  cropOffsetLeft: 0,
  cropOffsetTop: 0,
  size: 143109

**Comment by lovell** [comment]
Thanks, to start with, here's the "baseline" centre crop output:

![](https://github.com/lovell/sharp/assets/210965/a3159f09-7339-4ee5-8882-5b9d16b7fc05)

Here's the output when using an entropy calculation.

![](https://github.com/lovell/sharp/assets/210965/75cb4f10-e82f-4889-a788-7506c436e067)

The entropy of the left edge is higher than the right, likely due to the light reflections on the wine glasses.

If I flip the input image horizontally and try again, I see the right edge "win", so the calculation appears to be consistent.

![](https://github.com/lovell/sharp/assets/210965/3196764e-f8af-46fc-9f8a-2c57391e72d8)

If I blank out the wine glasses, the suspected source of entropy:

![](https://github.com/lovell/sharp/assets/210965/e7da691d-c00a-4184-8e33-687e9aaa176d)

Then the entropy crop correctly removes this edge and produces:

![out-be](https://github.com/lovell/sharp/assets/210965/8aff13a9-983f-4cf7-b201-2eb1ab2f5ea4)

**Comment by davidrinnan** [comment]
first thank you. second this is so sick. when I implemented this I truly had issues with entropy. tried many different images (images I have now tried that didnt work but now does work) I then tested a new setup with lambda where I used a layer with sharp. 
https://serverlessrepo.aws.amazon.com/applications/us-east-1/987481058235/nodejs-sharp-lambda-layer

But silly enough I just kept testing the same file which had issues in itself, as you pointed out. 

So basically once I used that sharp layer linked above I had a working version.

**Comment by lovell** [comment]
No worries, hope everyone enjoyed their wine tasting, skål!
